index.js

```diff
@@ -0,0 +1,116 @@
+"use strict";
+
+require('core-js/shim');
+
+var P = require('bluebird');
+var dgram = P.promisifyAll(require('dgram'));
+var jsPack = require('jspack').jspack;
+
+/**
+ * Creates a new cache purger instance
+ *
+ * @param options object containing options for a cache purger:
+ *  - log:    logging function (default no-op)
+ *  - routes: array of route objects to map a resource url to the cache endpoint
+ *      - rule: ether regex for a resource url, or 'undefined' if it's a default endpoint
+ *      - host: cache endpoint host
+ *      - port: cache endpount port
+ *  - multicast_ttl: standard UDP multicast TTL option (default 8)
+ * @constructor
+ */
+function HTCPPurger(options) {
+    var self = this;
+    self.options = options || {};
+    self.log = self.options.log || function() {};
+
+    if (!self.options.routes) {
+       throw new Error('Config error. At least one route must be specified');
+    }
+
+    self.options.routes.forEach(function(routeSpec) {
+        if (routeSpec.rule && /^\/.+\/$/.test(routeSpec.rule)) {
+            var regExp = new RegExp(routeSpec.rule.substring(1, routeSpec.rule.length - 1));
+            routeSpec.rule = function(url) {
+                return regExp.test(url);
+            };
+        } else {
+            routeSpec.rule = function() { return true; };
+        }
+    });
+
+    self.options.multicast_ttl = self.options.multicast_ttl || 8;
+
+    self.seqReqId = 1;
+}
+
+/**
+ * Construct a UDP datagram with HTCP packet for Varnish flush of the url
+ * @param url a url of the resource that should be flushed
+ * @returns {Buffer} resulting HTCP packet bytes
+ * @private
+ */
+HTCPPurger.prototype._constructHTCPRequest = function(url) {
+    var self = this;
+    var htcpSpecifier = jsPack.Pack('!H4sH' + url.length + 'sH8sH',
+        [4, 'HEAD', url.length, url, 8, 'HTTP/1.0', 0]);
```

Comment:
`url.length` would be the string length, but I think you want the byte length here. You can get that by converting the url to a Buffer, and then taking its length.

An alternative to jspack would be to use https://nodejs.org/api/buffer.html methods to build this up. Those are native in nodejs, and might perform better.